### PR TITLE
Commit to Spring Boot Security Branch First

### PR DESCRIPTION
**Pull request by romaxa11**

### CODE REVIEW

File: src/main/java/mate/academy/intro/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth
+                                .requestMatchers("/auth/**")
+                                .permitAll()
+                                .anyRequest()
+                                .authenticated())
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/config/SecurityConfig.java]: just a suggestion: let's permit all to see 
1. /error - error page
2. /swagger-ui/** - documentation of your API.

File: src/main/java/mate/academy/intro/exception/RegistrationException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.intro.exception;
+
+public class RegistrationException extends Exception {
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/exception/RegistrationException.java]: let's make this exception extended from RuntimeException to reduce boilerplate code. 
```suggestion
public class RegistrationException extends RuntimeException {
```

File: src/main/java/mate/academy/intro/service/user/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private final RoleService roleService;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequest request) throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable to complete registration");
+        }
+        if (!request.getPassword().equals(request.getRepeatPassword())) {
+            throw new RegistrationException("Wrong password or repeatPassword...");
+        }
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/service/user/impl/UserServiceImpl.java]: You already have checked that password and repeatPassword are the same using @FieldMatch. 

```suggestion
```

File: src/main/java/mate/academy/intro/model/Role.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/model/Role.java]: @Data is not recommended for use with Entity.

File: src/main/java/mate/academy/intro/exception/RegistrationException.java (hunk 2 above)
Comment (sokimaaa) [reply src/main/java/mate/academy/intro/exception/RegistrationException.java]: make sure you handle this exception using GlobalExceptionHandler. 
I didn't find that you handled it.

File: src/main/java/mate/academy/intro/validation/Email.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = EmailValidator.class)
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Email {
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/validation/Email.java]: you do not use this annotation, so, you can delete it

File: src/main/java/mate/academy/intro/model/User.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import jakarta.persistence.Table;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Data
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/model/User.java]: @DaTa is not recommended for use with Entity.

File: src/main/java/mate/academy/intro/validation/EmailValidator.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.intro.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class EmailValidator implements ConstraintValidator<Email,String> {
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/validation/EmailValidator.java]: you can delete this class as well

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/intro/model/Book.java [hunk 8]
```diff
@@ -7,12 +7,18 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.Where;
 
+@Getter
+@Setter
+@ToString
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/model/Book.java:19]: why do you need @ToString? delete it anywhere

**Review by Rommelua — APPROVED** [review]

**Pull request merged**